guides/source/development_dependencies_install.md

```diff
@@ -43,215 +43,121 @@ $ git clone https://github.com/rails/rails.git
-```
-
-If you are on Arch Linux, you will need to run:
-
-```bash
-$ sudo pacman -S sqlite
-```
-
-For FreeBSD users, you're done with:
-
-```bash
-# pkg install sqlite3
-```
-
-Or compile the `databases/sqlite3` port.
+* Action Cable depends on Redis
+* Active Record depends on SQLite3, MySQL and PostgreSQL
+* Active Storage depends on Yarn (additionally Yarn depends on [Node.js](https://nodejs.org/))
+* Active Support depends on memcached and Redis
+* Railties depend on a JavaScript runtime environment, such as having [Node.js](https://nodejs.org/) installed.
 
-Get a recent version of [Bundler](https://bundler.io/)
+Install all the services you need to properly test the full gem you'll be making changes to.
+For example, if you will make changes to Action Cable, you _must_ install Redis, in order to be able to run all Action Cable tests before submitting your pull request.
 
-```bash
-$ gem install bundler
-$ gem update bundler
-```
+NOTE: Redis' documentation discourage installations with package managers as those are usually outdated. Installing from source and bringing the server up is straight forward and well documented on [Redis' documentation](https://redis.io/download#installation).
 
-and run:
+NOTE: Active Record's test suite runs three times: once for SQLite3, once for MySQL, and once for PostgreSQL. We are going to see now how to set up the environment for them.
 
-```bash
-$ bundle install --without db
-```
+NOTE: Extracting previews, tested in Active Storage's test suite requires third-party applications:
+ImageMagick for images, FFmpeg for video and muPDF for PDFs, and on macOS also XQuartz and Poppler.
+Without these applications installed, Active Storage tests will raise errors.
 
-This command will install all dependencies except the MySQL and PostgreSQL Ruby drivers. We will come back to these soon.
+WARNING: If you're working with Active Record code, you _must_ ensure that the tests pass for at least MySQL, PostgreSQL, and SQLite3. Subtle differences between the various adapters have been behind the rejection of many patches that looked OK when tested only against single adapter.
 
-NOTE: If you would like to run the tests that use memcached, you need to ensure that you have it installed and running.
+Below you can find instructions on how to install all of the additional tools for different OSes.
 
-You can use [Homebrew](https://brew.sh/) to install memcached on macOS:
+#### macOS
 
-```bash
-$ brew install memcached
-```
-
-On Ubuntu you can install it with apt-get:
-
-```bash
-$ sudo apt-get install memcached
-```
-
-Or use yum on Fedora or CentOS:
-
-```bash
-$ sudo yum install memcached
-```
+On macOS you can use [Homebrew](https://brew.sh/) to install all of the additional tools.
 
-If you are running on Arch Linux:
+To install all run:
 
 ```bash
-$ sudo pacman -S memcached
+$ brew install sqlite3 mysql postgresql redis memcached yarn imagemagick ffmpeg mupdf-tools poppler
```

Comment:
If you are inside the Rails repository you can run just `brew bundle` and it will take care of everything.

https://github.com/rails/rails/blob/master/Brewfile

In fact I prefer that to this list here that will get outdated soon.

Reply:
Thanks for feedback!

I mentioned that in the PR description above. I've replaced this code snippet with `brew bundle`. Please have a look.